test/quantization/test_backward_compatibility.py

```diff
@@ -0,0 +1,159 @@
+import torch.nn.intrinsic.quantized as nniq
+
+# Testing utils
+from torch.testing._internal.common_utils import TestCase
+from torch.testing._internal.common_utils import run_tests
+
+# TODO: remove this global setting
+# JIT tests use double as the default dtype
+torch.set_default_dtype(torch.double)
+
+class TestSerialization(TestCase):
+    # Copy and modified from TestCase.assertExpected
+    def _test_op(self, qmodule, subname=None, prec=None):
+        r""" Test quantized modules serialized previously can be loaded
+        with current code, make sure we don't break backward compatibility for the
+        serialization of quantized modules
+        """
+        def remove_prefix(text, prefix):
+            if text.startswith(prefix):
+                return text[len(prefix):]
+            return text
+        # NB: we take __file__ from the module that defined the test
+        # class, so we place the expect directory where the test script
+        # lives, NOT where test/common_utils.py lives.
+        module_id = self.__class__.__module__
+        munged_id = remove_prefix(self.id(), module_id + ".")
+        test_file = os.path.realpath(sys.modules[module_id].__file__)
+        # TODO: change to quantization/serialized after we add test_quantization.py
+        # under pytorch/test folder
+        base_name = os.path.join(os.path.dirname(test_file),
+                                 "serialized",
+                                 munged_id)
+
+        subname_output = ""
+        if subname:
+            base_name += "_" + subname
+            subname_output = " ({})".format(subname)
+
+        input_file = base_name + ".input.pt"
+        state_dict_file = base_name + ".state_dict.pt"
+        scripted_module_file = base_name + ".scripted.pt"
+        traced_module_file = base_name + ".traced.pt"
+        expected_file = base_name + ".expected.pt"
+
+        # saving code : uncomment when adding a new test
+        # and run the new test, e.g.
+        # python test/quantization/test_backward_compatibility.py TestSerialization.test_conv3d
+        data = torch.randn(2, 3).float()
+        data = torch.quantize_per_tensor(data, 0.5, 2, torch.quint8)
+        torch.save(data, input_file)
+        torch.save(qmodule.state_dict(), state_dict_file)
+        torch.jit.save(torch.jit.script(qmodule), scripted_module_file)
+        torch.jit.save(torch.jit.trace(qmodule, data), traced_module_file)
+        torch.save(qmodule(data), expected_file)
+
+        data = torch.load(input_file)
+        qmodule.load_state_dict(torch.load(state_dict_file))
+        qmodule_scripted = torch.jit.load(scripted_module_file)
+        qmodule_traced = torch.jit.load(traced_module_file)
+
+        expected = torch.load(expected_file)
+        self.assertEqual(qmodule(data), expected, prec=prec)
+        self.assertEqual(qmodule_scripted(data), expected, prec=prec)
+        # self.assertEqual(qmodule_traced(data), expected, prec=prec)
```

Comment:
Why is this commented?  We should be able to run a traced model:
https://github.com/pytorch/pytorch/blob/3c85f44ce8335753ecf9d33307a013c9d98c6509/torch/testing/_internal/common_quantization.py#L139
Infact, is the restriction referred to in this code still needed? 
https://github.com/pytorch/pytorch/issues/23986 is closed, so this should work.

Reply:
check_trace is removed in the most recent commit